### PR TITLE
Testnet scripts: separate options to start Proxy and TxGen

### PR DESCRIPTION
**Pull request by camilbancioiu**
This PR adds the following improvements to the local testnet scripts:
* the previous `$PRIVATE_REPOS` variable is replaced by `$USE_PROXY` and `$USE_TXGEN`, which can now be enabled/disabled independently
* added variable `$ALWAYS_BUILD_ARWEN`, which controls whether Arwen is rebuilt on every testnet start (leave it set to `1` unless you really know what you are doing)
* added new scripts `pause.sh` and `resume.sh`, which send the `SIGSTOP` and `SIGCONT` signals respectively to all the processes started by the testnet (seednode, validators, observers, proxy, txgen); useful to suspend a running testnet, reducing the load on the computer while you work on the source code
* defined file `local.sh` in the same folder as the rest of the scripts (must be created manually), which is imported by `variables.sh` but *ignored* by Git, allowing easy local overrides without generating noise in the repository; use it to temporarily override any options in `variables.sh`, such as network topology or custom profiles
* added script `reset.sh`, which stops a running testnet, deletes its folder, creates it anew, regenerates configuration and then starts it back; useful for a full reset while the testnet is still active
* some minor refactoring

### CODE REVIEW

File: scripts/testnet/config.sh [hunk 1]
```diff
@@ -20,11 +20,14 @@ updateSeednodeConfig
...
-  prepareFolders_PrivateRepos
-
+if [ $USE_PROXY -eq 1 ]; then
+  prepareFolders_Proxy
   copyProxyConfig
   updateProxyConfig
+fi
+
+if [ $USE_TXGEN -eq 1 ]; then
+	prepareFolders_TxGen
```
Comment (AdoAdoAdo) [inline scripts/testnet/config.sh:30]: indentation?

File: scripts/testnet/start.sh [hunk 2]
```diff
@@ -46,24 +46,31 @@ showTerminalSession "elrond-nodes"
...
-if [ $PRIVATE_REPOS -eq 1 ]; then
-  # Phase 5: build the Proxy and TxGen
-  prepareFolders_PrivateRepos
+# Phase 5: build the Proxy and TxGen
+if [ $USE_PROXY -eq 1 ]; then
+  prepareFolders_Proxy
   buildProxy
+fi
+if [ $USE_TXGEN -eq 1 ]; then
+	prepareFolders_TxGen
```
Comment (AdoAdoAdo) [inline scripts/testnet/start.sh:55]: indentation

**Comment by camilbancioiu** [comment]
Please ignore this PR for now, it was opened prematurely. There will be new changes pushed into it, as development of `erdjs` progresses (the scripts will be improved soon, e.g. with functionality to control a running testnet without stopping it, by pausing all node/proxy/txgen/seednode processes, then resuming them).

File: scripts/testnet/config.sh (hunk 1 above)
Comment (camilbancioiu) [reply scripts/testnet/config.sh]: Fixed.

File: scripts/testnet/start.sh (hunk 2 above)
Comment (camilbancioiu) [reply scripts/testnet/start.sh]: Fixed.

**Pull request merged**